synapse/handlers/device.py

```diff
@@ -505,8 +505,9 @@ async def notify_device_update(
             "device_list_key", position, users={user_id}, rooms=room_ids
         )
 
-        # We may need to do some processing asynchronously.
-        self._handle_new_device_update_async()
+        # We may need to do some processing asynchronously for local user IDs.
+        if self.hs.is_mine_id(user_id):
+            self._handle_new_device_update_async()
```

Comment:
Don't fully understand the machinery here, but the docstring of `_handle_new_device_update_async` says:

> Called when we have a new local device list update that we need to send out over federation.

So that seems reasonable.